src/taoensso/timbre/appenders/core.cljc

```diff
@@ -84,25 +99,11 @@
 
      {:enabled? true
       :fn
-      (let [lock (Object.)]
-        (fn self [data]
-          (let [{:keys [output_]} data]
-            (let [output (force output_)]
-              (try
-                (when locking? (monitor-enter lock)) ; For thread safety, Ref. #251
-                (with-open [^java.io.BufferedWriter w (jio/writer fname :append append?)]
-                  (.write   w ^String output)
-                  (.newLine w))
-
-                (catch java.io.IOException e
-                  (if (:spit-appender/retry? data)
-                    (throw e) ; Unexpected error
-                    (do
-                      (jio/make-parents fname)
-                      (self (assoc data :spit-appender/retry? true)))))
-                (finally
-                  (when locking?
-                    (monitor-exit lock))))))))}))
+      (fn self [data]
+        (if locking?
+          (locking (Object.)
```

Comment:
The object to lock on needs to be listed out into a let here @ericdallo, since every invocation of the function needs to be locked on the same object. I think this is what @ptaoussanis referred to in his latest comment.

Reply:
thanks, I'll change it

Reply:
I think it's fixed now @ptaoussanis